broker/api.py

```diff
@@ -407,3 +430,60 @@ def provision_cdn_instance(instance_id: str, domain_names: list, params: dict):
     else:
         instance.origin_protocol_policy = CDNServiceInstance.ProtocolPolicy.HTTPS.value
     return instance
+
+
+def validate_migration_to_cdn_params(params):
+    required = [
+        "origin",
+        "path",
+        "forwarded_cookies",
+        "forward_cookie_policy",
+        "forwarded_headers",
+        "insecure_origin",
+        "error_responses",
+        "cloudfront_distribution_id",
+        "cloudfront_distribution_arn",
+        "iam_server_certificate_name",
+        "iam_server_certificate_id",
+        "iam_server_certificate_arn",
+        "domain_internal",
+    ]
+    for param in required:
+        # since this should only be hit by another app, it seems
+        # fair and smart to require all params
+        if not param in params:
+            raise ClientError(f"Missing parameter {param}")
+
+
+def update_cdn_params_for_migration(instance, params):
+    instance.cloudfront_origin_hostname = params["origin"]
+    instance.cloudfront_origin_path = params["path"]
+    instance.forwarded_cookies = params["forwarded_cookies"]
+    instance.forward_cookie_policy = params["forward_cookie_policy"]
+    if params["insecure_origin"]:
+        instance.origin_protocol_policy = CDNServiceInstance.ProtocolPolicy.HTTP.value
+    else:
+        instance.origin_protocol_policy = CDNServiceInstance.ProtocolPolicy.HTTPS.value
+    instance.forwarded_headers = params["forwarded_headers"]
+    instance.error_responses = params["error_responses"]
+    instance.cloudfront_distribution_id = params["cloudfront_distribution_id"]
+    instance.cloudfront_distribution_arn = params["cloudfront_distribution_arn"]
+    instance.domain_internal = params["domain_internal"]
+    instance.current_certificate = Certificate(service_instance_id=instance.id)
+    instance.current_certificate.iam_server_certificate_id = params[
+        "iam_server_certificate_id"
+    ]
+    instance.current_certificate.iam_server_certificate_arn = params[
+        "iam_server_certificate_arn"
+    ]
+    instance.current_certificate.iam_server_certificate_name = params[
+        "iam_server_certificate_name"
+    ]
```

Comment:
Just wanted to check to make sure nothing else had to happen here at the end after setting all of these things!

Reply:
nope - because this is an object passed by reference, this is just about updating its params. The only reason I pulled it out into a function was so the if-elif-else chain above would read more easily

Reply:
Makes sense!